OpenRA.Game/Primitives/PriorityQueue.cs

```diff
@@ -91,6 +94,33 @@ public T Pop()
 			return ret;
 		}
 
+		public bool TryPeek(out T item)
+		{
+			if (level == 0)
+			{
+				item = default(T);
+				return false;
+			}
+
+			item = At(0, 0);
+			return true;
+		}
+
+		public bool TryPop(out T item)
+		{
+			if (level == 0)
```

Comment:
`Pop` also checks `index`. Is it safe to ignore that here?

Reply:
`Empty` only uses `level`. So I assumed it would be safe.